### PR TITLE
Use `MaxPool-14` to remove Torch Ceil workaround from PT FE

### PR DESCRIPTION
**Pull request by p-wysocki**
### Tickets:
 - CVS-122418

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_pooling.py [hunk 1]
```diff
@@ -143,15 +142,7 @@ def test_avg_pool1d(self, params, ceil_mode, count_include_pad, ie_device, preci
...
-    @pytest.mark.parametrize(
-        "params",
-        d2_params
-        + [
-            pytest.param(
-                {"kernel_size": [8, 8], "stride": [8, 4], "padding": 1},
-                marks=pytest.mark.xfail(reason="Sliding windows that would start in the right padded are ignored.")
-            )
-        ])
+    @pytest.mark.parametrize("params", d2_params)
```
Comment (mitruska) [inline tests/layer_tests/pytorch_tests/test_pooling.py:145]: Please update only MaxPool tests within this PR

File: src/frontends/pytorch/src/op/avg_poolnd.cpp [hunk 2]
```diff
@@ -68,7 +68,7 @@ OutputVector translate_avg_poolnd(const NodeContext& context) {
     }
 
     return {context.mark_node(
-        std::make_shared<v1::AvgPool>(input, strides, pads, pads, kernel, !count_include_pad, rounding_type))};
+        std::make_shared<v14::AvgPool>(input, strides, pads, pads, kernel, !count_include_pad, rounding_type))};
```
Comment (mitruska) [inline src/frontends/pytorch/src/op/avg_poolnd.cpp:71]: Please remove AvgPool changes from this PR

**Review by mitruska — COMMENTED** [review]
As we have agreed to provide only MaxPool changes within this PR, 
please remove AvgPool leftovers.

File: src/frontends/pytorch/src/op/avg_poolnd.cpp (hunk 2 above)
Comment (p-wysocki) [reply src/frontends/pytorch/src/op/avg_poolnd.cpp]: Done.

File: src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp [hunk 3]
```diff
@@ -460,6 +461,17 @@ void Transformations::PreLpt(const std::vector<ov::element::Type>& defaultPrecis
...
         ov::pass::ConvertBatchToSpace, ov::pass::ConvertSpaceToBatch);
 
+    CPU_SET_CALLBACK_COMMON(
+        manager,
+        [](const_node_ptr& node) -> bool {
+            const auto maxpool = std::dynamic_pointer_cast<const ov::op::v14::MaxPool>(node);
+            if (maxpool && maxpool->get_rounding_type() != ov::op::RoundingType::CEIL_TORCH) {
+                return false;
+            }
+            return true;
```
Comment (praasz) [inline src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp:471]:
```suggestion
            return !maxpool ||  maxpool->get_rounding_type() == ov::op::RoundingType::CEIL_TORCH;
```

**Review by praasz — APPROVED** [review]

**Review by mitruska — APPROVED** [review]

**Review by mmikolajcz — APPROVED** [review]

**Pull request merged**